File: src/main/java/com/bank/service/TransactionService.java
package com.bank.service;

import com.bank.dto.DepositRequest;
import com.bank.dto.TransactionResponse;
import com.bank.dto.TransferRequest;
import com.bank.dto.WithdrawalRequest;
import org.springframework.data.domain.Page;

public interface TransactionService {
    TransactionResponse deposit(Long accountId, String userEmail, DepositRequest request);
    TransactionResponse withdraw(Long accountId, String userEmail, WithdrawalRequest request);
    TransactionResponse transfer(String userEmail, TransferRequest request);
    Page<TransactionResponse> getUserTransactions(String userEmail, int page, int size);
}

File: frontend/src/services/transactionService.js
import API from './api';

export const transactionService = {
  deposit: (accountId, data) => API.post(`/accounts/${accountId}/deposit`, data),
  withdraw: (accountId, data) => API.post(`/accounts/${accountId}/withdraw`, data),
  transfer: (data) => API.post('/transactions/transfer', data),
  getTransactions: (page = 0, size = 10) => API.get(`/transactions?page=${page}&size=${size}`),
};
